### DISCUSSION TITLE
Asset URLs not using ASSET_URL environment variable

### DISCUSSION THREAD

**Discussion by fh32000** [laravel/nova-issues#5382] [Q&A] [answered]
- Laravel Version: 10.1
- Nova Version: 4.22.0
- PHP Version:8.1
- Database Driver & Version: mysql server 8.0
- Operating System and Version: ubuntu:22.04
- Browser type and version: Google Chrome Version 109.0.5414.119 
- Reproduction Repository: null

I'm having trouble getting Laravel Nova to use the ASSET_URL environment variable for generating asset URLs. I've set ASSET_URL to point to my S3 bucket and uploaded the contents of my public folder, along with the vendor/nova folder, to S3.
```
ASSET_URL=https://my-bucket.s3.amazonaws.com
```
However, when I load my application, the asset URLs still use the APP_URL environment variable instead of ASSET_URL. I've tried updating my config/filesystems.php file to include a new s3 disk with the url option set to ASSET_URL, and setting the asset_disk option in my config/nova.php file to s3, but the issue persists.
```
// config/filesystems.php
'disks' => [
    // ...
    's3' => [
        'driver' => 's3',
        'key' => env('AWS_ACCESS_KEY_ID'),
        'secret' => env('AWS_SECRET_ACCESS_KEY'),
        'region' => env('AWS_DEFAULT_REGION'),
        'bucket' => env('AWS_BUCKET'),
        'url' => env('ASSET_URL'),
        'endpoint' => env('AWS_ENDPOINT'),
    ],
],


// config/nova.php
'asset_disk' => 's3',

````

I've also cleared my application cache and checked the asset URLs in my browser's developer console, but no luck.

Could someone help me troubleshoot this issue and ensure that Laravel Nova is using my ASSET_URL environment variable for asset URLs? Any help would be greatly appreciated.

Thank you!

**Comment by crynobone** [comment, marked as answer]
Laravel Nova uses `mix()` helper under the hood and would rely on the following:

https://github.com/laravel/nova-dusk-suite/blob/97ce3ed41ca84cfcd5caa60255b8d73b4d74f698/config/app.php#L61

**Reply by fh32000** [reply]
Thank you it worked great
in config/app.php i add
```
'mix_url' => env('MIX_ASSET_URL', env('ASSET_URL', null)),
```
